controller/api/destination/endpoint_profile_translator.go

```diff
@@ -124,10 +124,11 @@ func (ept *endpointProfileTranslator) update(address *watcher.Address) {
 		return
 	}
 
+	_, opaqueProtocol := opaquePorts[address.Port]
 	profile := &pb.DestinationProfile{
 		RetryBudget:    defaultRetryBudget(),
 		Endpoint:       endpoint,
-		OpaqueProtocol: address.OpaqueProtocol,
+		OpaqueProtocol: opaqueProtocol || address.OpaqueProtocol,
```

Comment:
Would it be possible to just rely on `address.OpaqueProtocol` here, and refactor `ept.createEndpoint()` so it also relies solely on `address.OpaqueProtocol`?

Reply:
this would be a good change but after giving it a try, I think it's a larger refactor that would be better to separate from this change.  basically, we not consistent with where we're consuming opaque protocol annotations and we read them both in the translator and in the watcher.  refactoring this to centralize this so that opaque protocol annotations are consumed in one place would be a good refactor.